### PR TITLE
Automate backwards compatibility check

### PR DESCRIPTION
**Pull request by maxmarkus**
### `lerna bootstrap` required

Created regression testing script to check out and run compatibility tests automatically. 
`npm run test:compatibility`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
 
 When the application is ready:
 
-- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
 - Run `npm run e2e:open` in the `core/examples/luigi-sample-angular` folder to start tests in the interactive mode.
+- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
+
+### Backwards Compatibility Tests
+
+To ensure that applications written for previous versions of Luigi still work after Luigi gets updated via npm. Before running the tests, start bundle Luigi by running `lerna run bundle` in the main repo folder.
```
Comment (JohannesDoberer) [inline README.md]: main repo means root or in core folder?

File: README.md [hunk 2]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
 
 All projects in the repository use [Prettier](https://prettier.io) to format source code. Run the `npm install` command in the root folder to install it along with [husky](https://github.com/typicode/husky), the Git hooks manager. Both tools ensure proper codebase formatting before committing it.
 
-### Tests
+### Unit Tests
```
Comment (bszwarc) [inline README.md]:
```suggestion
### Unit tests
```

File: README.md [hunk 3]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
+
+To ensure that existing features still work as expected after your changes, run unit tests by running `npm run test` the [core](/core) folder.
+
+### E2E Tests
```
Comment (bszwarc) [inline README.md]:
```suggestion
### E2E tests
```

File: README.md [hunk 4]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
 
 All projects in the repository use [Prettier](https://prettier.io) to format source code. Run the `npm install` command in the root folder to install it along with [husky](https://github.com/typicode/husky), the Git hooks manager. Both tools ensure proper codebase formatting before committing it.
 
-### Tests
+### Unit Tests
+
+To ensure that existing features still work as expected after your changes, run unit tests by running `npm run test` the [core](/core) folder.
```
Comment (bszwarc) [inline README.md]:
```suggestion
To ensure that existing features still work as expected after your changes, run unit tests using the `npm run test` command in the [core](/core) folder.
```

File: README.md [hunk 5]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
 
 To ensure that existing features still work as expected after your changes, run UI tests from the [Angular example application](/core/examples/luigi-sample-angular). Before running the tests, start the sample application by using the `npm start` command in the application folder.
 
 When the application is ready:
 
-- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
 - Run `npm run e2e:open` in the `core/examples/luigi-sample-angular` folder to start tests in the interactive mode.
+- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
+
+### Backwards Compatibility Tests
```
Comment (bszwarc) [inline README.md]:
```suggestion
### Backward compatibility tests
```

File: README.md (hunk 1 above)
Comment (bszwarc) [inline README.md]:
```suggestion
Use these tests to ensure that applications written for previous versions of Luigi still work after Luigi gets updated with npm. Before running the tests, bundle Luigi by running `lerna run bundle` in the main repository folder.
```

File: README.md [hunk 6]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
+
+To ensure that applications written for previous versions of Luigi still work after Luigi gets updated via npm. Before running the tests, start bundle Luigi by running `lerna run bundle` in the main repo folder.
+
+Install the json command line parser with `brew install jq` which is required for the script. Can be ommited if you run it with latest on the ci.
```
Comment (bszwarc) [inline README.md]: Question: I assume that the json command line parser is required for the script to work.
The second part is somehow unclear to me - can the command be omitted if you run it - it means the script? 
Also `with latest on CI` - doesn't say what :)

File: README.md [hunk 7]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
+
+- Run `npm run test:compatibility` in the main repo folder to start regression testing, you will be prompted you select the previous version.
```
Comment (bszwarc) [inline README.md]:
```suggestion
- Run `npm run test:compatibility` in the main repository folder to start regression testing. The system will prompt you to select the previous version. 
```

File: README.md [hunk 8]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
+- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
+
+### Backwards Compatibility Tests
+
+To ensure that applications written for previous versions of Luigi still work after Luigi gets updated via npm. Before running the tests, start bundle Luigi by running `lerna run bundle` in the main repo folder.
+
+Install the json command line parser with `brew install jq` which is required for the script. Can be ommited if you run it with latest on the ci.
+
+- Run `npm run test:compatibility` in the main repo folder to start regression testing, you will be prompted you select the previous version. 
+- Run `npm run test:compatibility -- --tag latest` in the main repo folder to start regression testing with the last version preselected.
```
Comment (bszwarc) [inline README.md]:
```suggestion
- Run `npm run test:compatibility -- --tag latest` in the main repository folder to start regression testing with the last version preselected. 
```

File: README.md [hunk 9]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
+
+### Backwards Compatibility Tests
+
+To ensure that applications written for previous versions of Luigi still work after Luigi gets updated via npm. Before running the tests, start bundle Luigi by running `lerna run bundle` in the main repo folder.
+
+Install the json command line parser with `brew install jq` which is required for the script. Can be ommited if you run it with latest on the ci.
+
+- Run `npm run test:compatibility` in the main repo folder to start regression testing, you will be prompted you select the previous version. 
+- Run `npm run test:compatibility -- --tag latest` in the main repo folder to start regression testing with the last version preselected. 
+- On the CI, run `npm run test:compatibility -- --install --tag latest` in the main repo folder to start install dependencies, bundle Luigi and run the tests with the last version preselected.
```
Comment (bszwarc) [inline README.md]:
```suggestion
- On the CI, run `npm run test:compatibility -- --install --tag latest` in the main repository folder to install dependencies, bundle Luigi and run the tests with the last version preselected. 
```

File: README.md (hunk 6 above)
Comment (maxmarkus) [reply README.md]: Correct, the command line parser is required to work for the script. With _latest_ I was referring to something like: If you run `npm run test:compatibility -- --tag latest`, jq is not required.

File: README.md [hunk 10]
```diff
@@ -28,11 +28,25 @@ For details, see [Luigi documentation](docs/README.md).
...
 
-- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
 - Run `npm run e2e:open` in the `core/examples/luigi-sample-angular` folder to start tests in the interactive mode.
+- Run `npm run e2e:run` in the `core/examples/luigi-sample-angular` folder to start tests in the headless browser.
+
+### Backward compatibility tests
+
+Use these tests to ensure that applications written for previous versions of Luigi still work after Luigi gets updated with npm. Before running the tests, bundle Luigi by running `lerna run bundle` in the main repository folder.
+
+Install the json command line parser with `brew install jq` which is required for the script. Can be ommited if you run it with latest on the ci.
```
Comment (bszwarc) [inline README.md]:
```suggestion
Install [jq](https://stedolan.github.io/jq/) using the `brew install jq` command. It is required for the script to work, however, you can omit it if the command you are using to run your tests is tagged `latest`. 
```

**Pull request merged**